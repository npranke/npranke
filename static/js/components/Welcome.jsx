import { Link } from 'react-router-dom'
import React from 'react'

import { getLocationPageTitle, getSendEventHandler } from '../utils'
import Workbook from '../../img/icon-workbook.png'

function Welcome() {
    document.title = getLocationPageTitle('home')

    return (
        <main className="welcome">
            <div className="intro">
                Welcome! <br />
                My name is Nicole; I&#8217;m a software engineer.
            </div>
            <div className="button-workbook">
                <Link
                    onClick={ getSendEventHandler(
                        'welcome',
                        'navigate',
                        'workbook',
                    ) }
                    to="/workbook"
                    target="_self"
                    rel="noreferrer"
                >
                    <img
                        src={ Workbook }
                        className="icon icon-workbook"
                        alt="Workbook icon"
                    />
                    <span className="text-workbook">Workbook</span>
                </Link>
            </div>
        </main>
    )
}

export default Welcome
