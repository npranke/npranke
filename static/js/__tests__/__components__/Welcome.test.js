import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import Welcome from '../../components/Welcome'

Enzyme.configure({ adapter: new Adapter() })

describe('Welcome', () => {
    test('has intro text with welcome', () => {
        const welcome = shallow(<Welcome />)

        expect(
            welcome.find('.intro').text(),
        ).toContain('Welcome!')
    })

    test('has link to workbook', () => {
        const welcome = shallow(<Welcome />)

        expect(
            welcome.find('.button-workbook a').props().href,
        ).toEqual('/')
    })

    test('has alt text for workbook icon', () => {
        const welcome = shallow(<Welcome />)

        expect(
            welcome.find('.button-workbook img').props().alt,
        ).toEqual('Workbook icon')
    })
})

describe('Welcome snapshot', () => {
    test('matches snapshot', () => {
        const welcome = shallow(<Welcome />)

        expect(welcome).toMatchSnapshot()
    })
})
