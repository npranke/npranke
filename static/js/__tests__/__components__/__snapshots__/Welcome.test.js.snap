// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome snapshot matches snapshot 1`] = `
<div
  className="welcome"
  role="main"
>
  <div
    className="intro"
  >
    Welcome! 
    <br />
    My name is Nicole; Iâ€™m a software engineer.
  </div>
  <div
    className="button-workbook"
  >
    <a
      href="mock-base-url"
      onClick={[Function]}
      rel="noreferrer"
      target="_self"
      title="Workbook"
    >
      <img
        alt="Workbook icon"
        className="icon icon-workbook"
        src="image-mock"
      />
      <span
        className="text-workbook"
      >
        Workbook
      </span>
    </a>
  </div>
</div>
`;
